### DISCUSSION TITLE
pauses between each action and robot go backwards accuracy issue

### DISCUSSION THREAD

**Discussion by familynygood** [pybricks/support#1793] [Q&A]
I have a spike prime robot to do the following and noticed there is some pauses/wait time between each function call.
After I asked the robot to run both attachment motor and made the robot repeat the same action and return home. When it is going in reversed direction, it is backing up turning instead of going foward and turning.  Can you please provide some insights on the following codes to help me pinpoint the issue. 

```python
def test(baseRobot:baselinerobot):
    baseRobot.drive_by_distance(distance=300)
    baseRobot.curve_angle(radius=100, angle=30)
    baseRobot.drive_by_distance(distance=350)
    baseRobot.curve_angle(radius=100, angle=-30)
    baseRobot.drive_by_distance(distance=150)
    baseRobot.move_right_attachment_motor_by_degree(degrees=150)
    baseRobot.move_right_attachment_motor_by_degree(degrees=-150)
    baseRobot.drive_by_distance(distance=-150)
    baseRobot.curve_angle(radius=100, angle=30)
    baseRobot.drive_by_distance(distance=-350)
    baseRobot.curve_angle(radius=100, angle=-30)
    baseRobot.drive_by_distance(distance=-300)
```

**Comment by laurensvalk** [comment]
It looks like you have made your own commands, so it would be easier to help if we could see what those commands are doing.
